pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -30,33 +30,16 @@ import (
 
 // ClaimInfo holds information required
 // to prepare and unprepare a resource claim.
+// +k8s:deepcopy-gen=true
 type ClaimInfo struct {
-	sync.RWMutex
 	state.ClaimInfoState
 	// annotations is a mapping of container annotations per DRA plugin associated with
 	// a prepared resource
 	annotations map[string][]kubecontainer.Annotation
 	prepared    bool
 }
 
-func (info *ClaimInfo) addPodReference(podUID types.UID) {
-	info.Lock()
-	defer info.Unlock()
-
-	info.PodUIDs.Insert(string(podUID))
-}
-
-func (info *ClaimInfo) deletePodReference(podUID types.UID) {
-	info.Lock()
-	defer info.Unlock()
-
-	info.PodUIDs.Delete(string(podUID))
-}
-
-func (info *ClaimInfo) addCDIDevices(pluginName string, cdiDevices []string) error {
-	info.Lock()
-	defer info.Unlock()
-
+func (info *ClaimInfo) updateCDIDevices(pluginName string, cdiDevices []string) error {
```

Comment:
I would still call this "add" to be symmetrical with `addPodReferences()`
```suggestion
func (info *ClaimInfo) addCDIDevices(pluginName string, cdiDevices []string) error {
```